src/hotspot/cpu/x86/frame_x86.cpp

```diff
@@ -68,8 +68,11 @@ bool frame::safe_for_sender(JavaThread *thread) {
     return false;
   }
 
-  // unextended sp must be within the stack and above or equal sp
-  if (!thread->is_in_stack_range_incl(unextended_sp, sp)) {
+  // unextended sp must be within the stack
+  // Note: sp can be greater than unextended_sp in the case of
+  // interpreted -> interpreted calls that go through a method handle linker,
+  // since those pop the last argument (the appendix) from the stack.
+  if (!thread->is_in_full_stack_checked(unextended_sp)) {
```

Comment:
The condition can remain stricter. The following should work too. Could you please check?
```suggestion
  if (!thread->is_in_stack_range_incl(unextended_sp + Interpreter::stackElementSize, sp)) {
```

Reply:
The sanity test worked :)

Reply:
... and no regression in the other serviceability tests.

Reply:
Probably it's better to adapt the limit though (sorry).
Problem with this version is that it allows `unextended_sp` outside the address range reserved for the stack. Let's see what other reviewers think.
```suggestion
  if (!thread->is_in_stack_range_incl(unextended_sp, sp - Interpreter::stackElementSize)) {
```

Reply:
Good point, you're right. I could have recognized this myself.

Reply:
Tests work as expected.